rpc/client.test.ts

```diff
@@ -1,8 +1,28 @@
 import * as A from "../deps/std/testing/asserts.ts"
+import * as C from "../mod.ts"
 import * as T from "../test_util/mod.ts"
 import * as U from "../util/mod.ts"
 import * as known from "./known/mod.ts"
 import * as msg from "./messages.ts"
+import { Provider } from "./mod.ts"
+import { ProviderHandlerError, ProviderSendError } from "./provider/errors.ts"
+
+const createMockClient = () => {
+  let listener: C.rpc.ProviderListener<Error, Error>
+  const providerMockFactory: Provider = (_discoveryValue, clientListener) => {
+    listener = clientListener
```

Comment:
this is very coupled to how the `rpc.Client` calls the provider factory.
I'm open to more elegant suggestions to extract the client listener so we can emit testing messages/errors